src/elements/content-sidebar/ActivitySidebarFilter.js

```diff
@@ -9,29 +9,51 @@ import { FormattedMessage } from 'react-intl';
 import DropdownMenu, { MenuToggle } from '../../components/dropdown-menu';
 import PlainButton from '../../components/plain-button';
 import messages from './messages';
-import { COMMENT_STATUS_OPEN } from '../../constants';
+import {
+    ACTIVITY_FEED_ITEM_ALL,
+    ACTIVITY_FEED_ITEM_OPEN,
+    ACTIVITY_FEED_ITEM_RESOLVED,
+    ACTIVITY_FEED_ITEM_TASKS,
+    ACTIVITY_FILTER_TITLE_ALL_COMMENTS,
+    COMMENT_STATUS_OPEN,
+    COMMENT_STATUS_RESOLVED,
+} from '../../constants';
 import { Menu, SelectMenuItem } from '../../components/menu';
-import type { FeedItemStatus } from '../../common/types/feed';
+import type { ActivityFilterAllTitle, ActivityFilterStatus } from '../../common/types/feed';
 import './ActivitySidebarFilter.scss';
 
 type ActivitySidebarFilterProps = {
-    feedItemStatus?: FeedItemStatus,
-    onFeedItemStatusClick: (status?: FeedItemStatus) => void,
-};
-
-const ACTIVITY_FEED_ITEM_ALL = 'all';
-const ACTIVITY_FEED_ITEM_OPEN = COMMENT_STATUS_OPEN;
-
-const statuses = [ACTIVITY_FEED_ITEM_ALL, ACTIVITY_FEED_ITEM_OPEN];
-const statusMap = {
-    [ACTIVITY_FEED_ITEM_ALL]: { msg: messages.activitySidebarFilterOptionAll, val: undefined },
-    [ACTIVITY_FEED_ITEM_OPEN]: { msg: messages.activitySidebarFilterOptionOpen, val: COMMENT_STATUS_OPEN },
+    activityFilterAllTitle?: ActivityFilterAllTitle,
+    activityFilterOptions?: ActivityFilterStatus[],
+    feedItemStatus?: ActivityFilterStatus,
+    onFeedItemStatusClick: (status?: ActivityFilterStatus) => void,
 };
 
 function ActivitySidebarFilter({
+    activityFilterAllTitle = ACTIVITY_FILTER_TITLE_ALL_COMMENTS,
```

Comment:
This title prop was created to ensure that Canvas could maintain current functionality while providing Preview its needed functionality. For preview, instead of the title "All Comments" that Canvas uses, Preview will need to use the title "All Activity".

An alternative considered was determining the title used by if the filter options included tasks. However, I was concerned this might couple two different things that in the future we may not want as tightly coupled.